### ISSUE TITLE
Questions about how simple CSS selectors translate to Radium

### ISSUE THREAD

**Issue by stephmilovic** [FormidableLabs/radium#846]
I’m new to Radium and I’m finding it difficult to do some of the most simple CSS the Radium way. Here are some examples of where I’m getting hung up:
1. Descendant selector. I understand that there are methods to apply rules if there is a hover on the parent by setting up state logic, but I can’t seem to find a straight forward way to target elements of a parent. For example, any item with the `.anchor` class will be red. But if the `.anchor` is a descendant of a `.listItem` it should be blue. In css, I would simply do `.listItem .anchor { color: blue; }`. I can’t seem to find a simple way to do that. 
2. Is there a way to target a specific element, like a or li? Or does it always need a class?
3. What about the adjacent selector, is there a Radium way to handle that? For example, `.one + .two { color: blue; }`
4. How can I select direct children? For example, `.container > .ul { color: blue }`, only targeting .ul elements that are the direct children of container
5. What if I want to give two classes the same style? Ex: `.one, .two { color: blue }`

**Comment by Cryza** [comment]
I think you didn't really understand what Radium is all about. **Radium does not use classes at all.** It only uses inline styles and provides some helpers and wrappers to facilitate that.

That means every information you need to style a component must be available where you generate your style object. So wherever your element that would normally carry the `.anchor` class is created, you have to add the functionality to also be shown in a blue variant there, e.g. through a `variant` prop or a `color` prop that defaults to blue. The ListItem component would then take care of passing that prop to the Anchor so that it is shown in that different variant.

That is basically the solution to points 1 through 4. For 5, you can simply create one style object that is applied to both the elements that should look the same. Or if they only share a part of the styling, you can also use an array to concatenate different style objects that should be used, one containing the base style and the second object containing the special adaptations.

TL;DR: Forget about classes and selectors. They don't exist in Radium. Rather think about how you can create an inline style object from the props you get (and add the additional ones if required).

**Comment by alexlande** [comment]
Thanks @Cryza for the great answer here 🤘
